Update the github plugin with an automatic gh run watch

torbjornvatn opened the pull request.

Commit:
Update the github plugin with an automatic gh run watch

judoole commented:
<3

judoole reviewed (approved):
Looks very good! Nice!

Commit:
Add some logging before gh run watch

Commit:
Make is weird and sensitive to line breaks

Commit:
Remove some bash lint warnings from install.sh

Also move the -include line to the start of the local Makefile.
More idiomatic according to a style guide I read

Commit:
Extract the deploy logic into a separate bash script to make it more readable

judoole commented on plugins/github/deploy.sh:
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
```
Smart

torbjornvatn replied:
Except that it doesn't work, because it only downloads .mk files when you update brokkr in the project that uses this :/

torbjornvatn replied:
Will look into that

judoole replied:
Hmm... That is true. The Airflow one downloads within the `airflow.mk` https://github.com/unacast/brokkr/blob/master/plugins/airflow/airflow.mk#L94

But it isn't quite nice that either. Maybe if we downloaded the full "folder" instead. Although local paths, AND figuring out every file that is within the folder using curl agains https://github.com is perhaps a challenge.

Commit:
Add a step to download optional .sh files for plugins

torbjornvatn commented:
@judoole I think I solved it!

judoole commented on brokkr/brokkr.mk:
```diff
@@ -30,6 +30,9 @@ $(_BROKKR_PLUGINS_MK): $(_BROKKR_PLUGINS_SENTINEL)
 		curl --fail -s "$$url" -o $(_BROKKR_PLUGINS_DIR)/$${plugin_path}.mk; \
+		url="https://raw.githubusercontent.com/$(BROKKR_REPO)/$${plugin_version}/plugins/$${plugin_path}.sh"; \
+		echo "Downloading $${url}"; \
+		curl --fail -s "$$url" -o $(_BROKKR_PLUGINS_DIR)/$${plugin_path}.sh; \
```
Ha! Smart

judoole replied:
I like it

torbjornvatn replied:
How did I make a release agian?
Manually?

judoole replied:
Yeah, unfortunately no README or fancy routine. Create a release and use the branch, or merge with master and release from that.
